torch/csrc/jit/script/compiler.cpp

```diff
@@ -795,6 +798,62 @@ std::shared_ptr<SugaredValue> BuiltinFunction::call(
       loc, *m.graph(), symbol, inputs, attributes, true));
 }
 
+std::shared_ptr<SugaredValue> ForkValue::call(
+    SourceRange loc,
+    Method& m,
+    at::ArrayRef<NamedValue> inputs_,
+    at::ArrayRef<NamedValue> attributes,
+    const std::vector<std::shared_ptr<SugaredValue>> &blocks,
+    size_t n_binders) {
+  if (blocks.size() != 1) {
+    throw ErrorReport(loc) << "Expected a single ScriptModule as the first"
+                           << " argument to Fork()";
+  }
+  auto inputs = toValues(*m.graph(), inputs_);
+  if (inputs.size() != 1 || inputs.at(0)->type()->kind() != TypeKind::TupleType) {
+    throw ErrorReport(loc) << "Expected a single tuple argument to Fork()";
+  }
+  auto n = m.graph()->insertNode(m.graph()->create(prim::Fork, inputs, 1))
+                ->setSourceLocation(std::make_shared<SourceRange>(loc));
+  auto body_block = n->addBlock();
+
+  std::vector<NamedValue> block_inputs;
```

Comment:
Don't replicate schema-checking logic. This is pretty wrong (not handling kwargs, not handle defaults, not handling self or positional correctly).